extension/entitystore/extension.go

```diff
@@ -94,9 +94,13 @@ func (e *EntityStore) Start(ctx context.Context, host component.Host) error {
 	e.serviceprovider = newServiceProvider(e.mode, e.config.Region, &e.ec2Info, e.metadataprovider, getEC2Provider, ec2CredentialConfig, e.done, e.logger)
 	switch e.mode {
 	case config.ModeEC2:
-		e.ec2Info = *newEC2Info(e.metadataprovider, e.done, e.config.Region, e.logger)
+		e.ec2Info = *newEC2Info(e.metadataprovider, e.kubernetesMode, e.done, e.config.Region, e.logger)
 		go e.ec2Info.initEc2Info()
-		go e.serviceprovider.startServiceProvider()
+		// Instance metadata tags is not usable for EKS nodes
+		// https://github.com/kubernetes/cloud-provider-aws/issues/762
+		if e.kubernetesMode == "" {
```

Comment:
similar to the above -- shouldn't this be `if e.kubernetesMode != modeEKS` ? So that we can account for native k8s?

Reply:
I disabled it for all kubernetes environment because `serviceprovider` isn't used for any Kubernetes related logics. [GetMetricServiceNameAndSource](https://github.com/aws/amazon-cloudwatch-agent/blob/d2fcc0347e816f31845d1592f646fa53a52da20d/extension/entitystore/extension.go#L167) is the only public function that is used for `awsentity` processor and that is only referenced in EC2 specific processor logic: https://github.com/aws/amazon-cloudwatch-agent/blob/161cd5ba722e09ab0aa3b0a6d3f40e3a06e6aa1d/plugins/processors/awsentity/processor.go#L214